lib/sort_by_length.rb

```diff
@@ -1,7 +1,27 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^^2) as .each is O(n) and there are 2 .eachs' 
+# Space complexity: O(n) since we are creating a new array, memory that is required to store the new array will depend on the length of the array
```

Comment:
See below on time complexity, but can you see how to do this with O(1) space complexity.  Maybe looking at the textbook curriculum?